src/Protocol/ActivityPub/Receiver.php

```diff
@@ -644,16 +644,19 @@ public static function processActivity(array $activity, string $body = '', int $
 				return true;
 			}
 		} else {
-			$attributed_to = '';
+			$attributed_to = null;
 		}
 
 		// Test the provided signatures against the actor and "attributedTo"
 		if ($trust_source) {
-			if ($attributed_to !== false && $attributed_to !== '') {
+			if (!is_null($attributed_to)) {
```

Comment:
Because of `JsonLD::fetchElement(): string|null` the `$attributed_to` variable could only contain `string|null`. The previous check for this variable was changed in commit 5772c7021689c018d7d3fd1b77a32ef7ebb3aefe:

```diff
-			if (!empty($attributed_to) && $actor !== null) {
+			if ($attributed_to !== false && $attributed_to !== '') {
```

The comparison with `!== false` is obviously wrong, but an empty string would also be `empty`. So my suggestion would be:

```suggestion
			if ($attributed_to !== null && $attributed_to !== '') {
```

Reply:
We have to cover two scenarios: One is a message type that contains `attributedTo` and one that doesn't. Should a message contain `attributedTo` but had it set to '', then this was intentional. Also we would then try to refetch the item - and possibly fetch a valid item.

Reply:
Very important point. 👍 We should add a comment about this, maybe in `JsonLD::fetchElement()`. There we should also make sure that the method only returns `string|null`.

Reply:
The function `fetchElement` can return multiple types, it can be integer as well or any other type that JSON supports.

Reply:
Than we should fix the `@return` in the phpdoc for `JsonLD::fetchElement()` as well.

Reply:
What should be the appropriate value for "any type and null"?

Reply:
`@return mixed|null`

Reply:
Is changed